### ISSUE TITLE
Support CentOS Stream 10

### ISSUE THREAD

**Issue by elfosardo** [metal3-io/ironic-image#529]
CentOS Stream 10 is not officially out yet but bits are already available https://mirror.stream.centos.org/10-stream/ and can be used for testing.

Important changes include:

kernel 6.10 or more recent
Python 3.12

This heavily impacts ironic and, as a consequence, the ironic-image

Before moving forward with this some changes are needed in the ironic project and the entire Openstack ecosystem.
For example while Python 3.12 is already being tested, it's not officially supported yet, and there's still no plan to start using CS10 in CI, although discussions on migrating some important components are ongoing since a while, for example removing eventlet https://review.opendev.org/c/openstack/governance/+/902585

This issue should be used to track the progression on the CS10 migration for ironic-image, while keeping an eye on the progress in the ironic project itself.

/kind feature

**Comment by elfosardo** [comment]
The Epoxy development cycle for Openstack has just started and includes support for Python 3.12, which is now actively tested

**Comment by elfosardo** [comment]
I've opened a bug to ask inotify-tools build for EPEL10 https://bugzilla.redhat.com/show_bug.cgi?id=2331998

**Comment by elfosardo** [comment]
ironic upstream CI is still based on CS9 and AFAIK there is no will to migrate it during this development cycle, but it could change depending on the priorities
my suggestion is to hold on to this until we have some movement in ironic upstream

**Comment by elfosardo** [comment]
community has decided to move forward and test both CS9 and CS10 in parallel
I'll propose a configuration for the ironic-image that could work with both versions

**Comment by tuminoid** [comment]
> @tuminoid I'd rather keep this open until we actually switch to CS10 by default

Can we make another issue about making it the default? It is easier to track and reference things when they aren't used for many things over long span of time.

**Comment by Rozzii** [comment]
I am in favor of closing this and creating a new Issue related to making it default.

**Comment by elfosardo** [comment]
https://github.com/metal3-io/ironic-image/issues/697
